pkg/sriov/network_test.go

```diff
@@ -0,0 +1,554 @@
+package sriov
+
+import (
+	"fmt"
+	"testing"
+
+	srIovV1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	defaultNetName         = "sriovnet"
+	defaultNetNsName       = "testnamespace"
+	defaultNetTargetNsName = "targetns"
+	defaultNetResName      = "resname"
+)
+
+//nolint:funlen
+func TestPullNetwork(t *testing.T) {
+	generateNetwork := func(name, namespace string) *srIovV1.SriovNetwork {
+		return &srIovV1.SriovNetwork{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: srIovV1.SriovNetworkSpec{},
+		}
+	}
+
+	testCases := []struct {
+		networkName         string
+		networkNamespace    string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		client              bool
+	}{
+		{
+			networkName:         "test1",
+			networkNamespace:    "test-namespace",
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			client:              true,
+		},
+		{
+			networkName:         "test2",
+			networkNamespace:    "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "sriovnetwork object test2 doesn't exist in namespace test-namespace",
+			client:              true,
+		},
+		{
+			networkName:         "",
+			networkNamespace:    "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "sriovnetwork 'name' cannot be empty",
+			client:              true,
+		},
+		{
+			networkName:         "test3",
+			networkNamespace:    "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "sriovnetwork 'namespace' cannot be empty",
+			client:              true,
+		},
+		{
+			networkName:         "test3",
+			networkNamespace:    "test-namespace",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "sriovnetwork 'apiClient' cannot be empty",
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testNetwork := generateNetwork(testCase.networkName, testCase.networkNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testNetwork)
+		}
+
+		testSettings = nil
```

Comment:
I believe this isn't needed since its just declared as a pointer above.

Reply:
+1